packages/app/src/systems/Chains/eth/services/txEthToFuel.ts

```diff
@@ -124,85 +101,126 @@ export class TxEthToFuelService {
-          address: ethAsset.address as `0x${string}`,
+      const { ethWalletClient, fuelAddress, amount } = input;
+      if (fuelAddress && ethWalletClient) {
+        const fuelPortal = EthConnectorService.connectToFuelMessagePortal({
           walletClient: ethWalletClient,
         });
 
-        const approveTxHash = await erc20Token.write.approve(
-          [VITE_ETH_FUEL_ERC20_GATEWAY as `0x${string}`],
+        const txHash = await fuelPortal.write.depositETH(
+          [fuelAddress.toB256() as `0x${string}`],
           {
             value: BigInt(amount),
             account: ethWalletClient.account,
           }
         );
 
+        return txHash;
+      }
+    } catch (e) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      if ((e as any)?.code === 'ACTION_REJECTED') {
+        throw new Error('Wallet owner rejected this transaction.');
+      }
+
+      throw e;
+    }
+
+    return undefined;
+  }
+
+  static async startErc20(input: TxEthToFuelInputs['startErc20']) {
+    TxEthToFuelService.assertStartErc20(input);
+
+    try {
+      const {
+        ethWalletClient,
+        fuelAddress,
+        amount,
+        ethAsset,
+        ethPublicClient,
+      } = input;
+
+      if (
+        isErc20Address(ethAsset?.address) &&
+        ethWalletClient &&
+        fuelAddress &&
+        ethPublicClient
+      ) {
+        const fuelErc20Gateway = EthConnectorService.connectToFuelErc20Gateway({
+          walletClient: ethWalletClient,
+        });
+        const erc20Token = EthConnectorService.connectToErc20({
+          address: ethAsset?.address as `0x${string}`,
+          walletClient: ethWalletClient,
+        });
+
+        const approveTxHash = await erc20Token.write.approve([
+          VITE_ETH_FUEL_ERC20_GATEWAY,
+          amount,
+        ]);
+
+        // TODO: apply workaround logic to use waitTransactionReceipt together
         const approveTxHashReceipt =
           await ethPublicClient.getTransactionReceipt({ hash: approveTxHash });
+
         if (approveTxHashReceipt.status !== 'success') {
           throw new Error('Failed to approve Token for transfer');
         }
 
-        const fuelTokenId = 'fuel';
-        const depositTxHash = await fuelErc20Gateway.write.deposit(
-          [
-            fuelAddress.toB256() as `0x${string}`,
-            ethAsset.address,
-            fuelTokenId,
-          ],
-          {
-            value: BigInt(amount),
-            account: ethWalletClient.account,
-          }
-        );
-        const depositTxHashReceipt =
-          await ethPublicClient.getTransactionReceipt({ hash: depositTxHash });
-        if (depositTxHashReceipt.status !== 'success') {
-          throw new Error('Failed to deposit Token');
-        }
-      } else {
-        const fuelPortal = TxEthToFuelService.connectToFuelMessagePortal({
-          walletClient: ethWalletClient,
-        });
+        const depositTxHash = await fuelErc20Gateway.write.deposit([
```

Comment:
just a suggestion, but we could move the initialization of fuelErc20Gateway right above here.